### DISCUSSION TITLE
After login it keeps forever loading

### DISCUSSION THREAD

**Discussion by tuxPT** [KRTirtho/spotube#617] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After entering credentials the client keeps forever loading

![image](https://github.com/KRTirtho/spotube/assets/17715960/d9b4aeaa-1dac-4b35-918c-cc5cdb54f04c)
![image](https://github.com/KRTirtho/spotube/assets/17715960/84ac031f-df15-46aa-96da-8c73eb83f563)

> ------- DEVICE INFO -------
> name: Arch Linux
> version: null
> id: arch
> idLike: null
> versionCodename: null
> versionId: rolling
> prettyName: Arch Linux
> buildId: rolling
> variant: null
> variantId: null
> machineId: HIDDEN_FOR_PRIVACY
> ------- APP INFO -------
> environment: release
> version: 3.0.1
> appName: spotube
> buildNumber: 20
> packageName: spotube
> ---------- ERROR ----------
> org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
> ------- STACK TRACE -------
> #0      DBusClient._callMethod (package:dbus/src/dbus_client.dart:1102)
> <asynchronous suspension>
> #1      DBusClient.callMethod (package:dbus/src/dbus_client.dart:608)
> <asynchronous suspension>
> #2      DBusRemoteObjectManager.getManagedObjects (package:dbus/src/dbus_remote_object_manager.dart:80)
> <asynchronous suspension>
> #3      NetworkManagerClient.connect (package:nm/src/network_manager_client.dart:2351)
> <asynchronous suspension>
> #4      ConnectivityPlusLinuxPlugin.checkConnectivity (package:connectivity_plus/src/connectivity_plus_linux.dart:24)
> <asynchronous suspension>
> #5      FlQueryConnectivityPlusAdapter.isConnected (package:fl_query_connectivity_plus_adapter/fl_query_connectivity_plus_adapter.dart:10)
> <asynchronous suspension>
> #6      Query._operate (package:fl_query/src/core/query.dart:143)
> <asynchronous suspension>
> ------- CUSTOM INFO -------

### Expected Behavior

Showing the playlists, letting play something, ...

### Steps to reproduce

1. Open app
2. Entering the sp_dc and sp_key
3. Waiting to sync but never syncs

### Operating System

Archlinux+CachyOS

### Spotube version

v3.0.0 and v3.0.1

### Installation source

Arch User Repository (AUR)

### Additional information

It happens also at flapak and appimage versions. I use the Hyprland window manager with gnome-keyring and seahorse.

**Comment by KRTirtho** [comment]
You don't have [networkmanager](https://archlinux.org/packages/extra/x86_64/networkmanager/) installed 
Spotube depends on networkmanager to detect if user is connected to internet to determine retrying behavior on Network Errors

**Reply by tuxPT** [reply]
I was able to solve by running the NetworkManager service but I don't like NetworkManager. Can you replace this hard dependency by something more agnostic ?  Like for example pinging to cloudflare or quad9, 1.1.1.1 or 9.9.9.9

**Reply by KRTirtho** [reply]
Sorry an underlying plugin used by spotube uses it. Nothing can be done in this regard unless the plugin project migrates

**Reply by KRTirtho** [reply]
This is the plugin https://github.com/fluttercommunity/plus_plugins/tree/main/packages/connectivity_plus/connectivity_plus

**Reply by tuxPT** [reply]
https://github.com/fluttercommunity/plus_plugins/issues/1241#issuecomment-1483770198

**Reply by KRTirtho** [reply]
Actually we've moved away from connectivity_plus & instead using a pure dart library internet_connection_checker so that's no longer an issue
You can test it out using the nightly channel
